src/cmds/qsub.c

```diff
@@ -5642,8 +5644,13 @@ do_daemon_stuff(void)
 		err_op = "";
 
 		memcpy(&workset, &readset, sizeof(readset));
+
 		timeout.tv_usec = 0;
-		timeout.tv_sec = QSUB_DMN_TIMEOUT; /* since timeout gets reset on Linux */
+		/* since timeout gets reset on Linux */
+		if (fl_unlinked == 1)
+			timeout.tv_sec = QSUB_DMN_TIMEOUT_SHORT;
```

Comment:
Add a comment here that you're using a shorter timeout in order to allow any foreground qsub processes to finish up before you exit.